pkgs/applications/graphics/tesseract/get-language-hashes.sh

```diff
@@ -10,14 +14,29 @@
 
 set -e
 
-tessdataRev=${1:-3cf1e2df1fe1d1da29295c9ef0983796c7958b7d}
+tessdataRev=3cf1e2df1fe1d1da29295c9ef0983796c7958b7d
+
+for arg in "$@"; do
+    if [[ $arg == --local-langs ]]; then
+        localLangs=1
+    else
+        tessdataRev=$arg
+    fi
+done
+
+if [[ $localLangs ]]; then
+    langCodesExpr='builtins.attrNames (builtins.removeAttrs tesseractLanguages [ "recurseForDerivations" "all" ])'
+else
+    langCodes=$(echo $(curl -s https://github.com/tesseract-ocr/tessdata/tree/$tessdataRev \
+                       | grep -ohP "(?<=/)[^/]+?(?=\.traineddata)" | sort))
+    langCodesExpr="lib.splitString \" \" \"$langCodes\""
+fi
 
-nixSrc=$(sed "s/TESSDATA_REV/$tessdataRev/" <<'EOF'
+nixSrc=$(sed -e "s/TESSDATA_REV/$tessdataRev/" -e "s/LANGUAGE_CODES/$langCodesExpr/" <<'EOF'
```

Comment:
Is it just me, or would just using shell (and exporting the list from Nix using `concatStringsSep, if needed`) be simpler in the current approach?

Reply:
Could you clarify a bit what you mean by this?

Reply:
Yep, that's infinitely more elegant, thanks for pointing it out! [Fixed.](https://github.com/NixOS/nixpkgs/pull/52379/commits/01e3738ddc40fe331649bad50bf2a27ebe3f82d5)